contracts/DrawCalculator.sol

```diff
@@ -362,33 +367,34 @@ contract DrawCalculator is IDrawCalculator, Ownable {
     ///@param _randomNumberThisPick users random number for this Pick
     ///@param _winningRandomNumber The winning number for this draw
     ///@param _masks The pre-calculate bitmasks for the prizeDistributions
+    ///@param _matchCardinality The cardinality of the draw
     ///@return The position within the prize tier array (0 = top prize, 1 = runner-up prize, etc)
     function _calculateTierIndex(
         uint256 _randomNumberThisPick,
         uint256 _winningRandomNumber,
-        uint256[] memory _masks
+        uint256[] memory _masks,
+        uint8 _matchCardinality
     ) internal pure returns (uint8) {
+        
         uint8 numberOfMatches = 0;
-        uint8 masksLength = uint8(_masks.length);
         
         // main number matching loop
-        for (uint8 matchIndex = 0; matchIndex < masksLength; matchIndex++) {
+        for (uint8 matchIndex = 0; matchIndex < _matchCardinality; matchIndex++) {
```

Comment:
you still want to loop through the masks.length.  the match cardinality may not match the array length.